bundles/framework/announcements/tool/AnnouncementsTool.js

```diff
@@ -0,0 +1,367 @@
+            var me = this,
+                announcementsSelection = {};
+            var pluginValues = me.getPlugin().getSelectedAnnouncements();
+            if (pluginValues.announcements) {
+                announcementsSelection.announcements = pluginValues.announcements;
+            }
+            return announcementsSelection;
+        },
+
+        /**
+        * Get values.
+        * @method getValues
+        * @public
+        *
+        * @returns {Object} tool value object
+        */
+        getValues: function () {
+            var me = this;
+
+            if (me.state.enabled) {
+                var pluginConfig = { id: this.getTool().id, config: this.getPlugin().getConfig() };
+                var announcementsSelection = me._getAnnouncementsSelection();
+
+                if (announcementsSelection && !jQuery.isEmptyObject(announcementsSelection)) {
+                    pluginConfig.config.announcements = announcementsSelection.announcements;
+                }
+                return {
+                    configuration: {
+                        mapfull: {
+                            conf: {
+                                plugins: [pluginConfig]
+                            }
+                        }
+                    }
+                };
+            } else {
+                return null;
+            }
+        },
+
+        /**
+         * "Sends" an event, that is, notifies other components of something.
+         *
+         * @method _sendEvent
+         * @param {String} eventName the name of the event
+         * @param {Whatever} eventData the data we want to send with the event
+         */
+        _sendEvent: function (eventName, eventData) {
+            var eventBuilder = Oskari.eventBuilder(eventName),
+                evt;
+
+            if (eventBuilder) {
+                evt = eventBuilder(eventData);
+                this.__sandbox.notifyAll(evt);
+            }
+        },
+
+        /**
+        * Sends an event to notify interested parties that the announcements have changed.
+        *
+        * @method _sendAnnouncementsChangedEvent
+        * @param {Object} announcements the changed announcement
+        */
+        _sendAnnouncementsChangedEvent: function (announcements) {
+            this._sendEvent('Announcements.AnnouncementsChangedEvent', announcements);
+        },
+
+        /**
+         * Retrieves the item from the list which value matches the code given
+         * or null if not found on the list.
+         *
+         * @method _getItemByCode
+         * @param {String} code
+         * @param {Array[Object]} list
+         * @return {Object/null}
+         */
+        _getItemByCode: function (code, list) {
+            var listLen = list.length,
+                i;
+
+            for (i = 0; i < listLen; ++i) {
+                if (list[i].title === code) {
+                    return list[i];
+                }
+            }
+            return null;
```

Comment:
```suggestion
     return list.find(item => item.title === code)
```

Reply:
I don't see why this would need it's own function. It only obfuscates reading the code.

Reply:
I just took it from another plugin/tool code, I’ll refactor accordingly